src/tracer.js

```diff
@@ -254,23 +260,19 @@ export default class Tracer {
     }
 
     /**
-     *  Makes a concrete sampling decision for the ctx span context based on information
-     *  available to it's child span.
-     *
-     *  The ctx context's deferred mask is unset after this decision is made
+     * Reapplies sampling if the parent span isn't sampled
      *
      * @param ctx the span context
      * @param operationName the operation name for the child
      * @param tags tags to be applied by the sampler
      */
-    processDeferredSampling(ctx: SpanContext, operationName: string, tags: any) {
-        if (ctx.isDeferredSampling) {
-            if (this._sampler.isSampled(operationName, tags)) {
-                ctx._flags |= constants.SAMPLED_MASK;
-            } else {
-                ctx._flags &= ~constants.SAMPLED_MASK;
-            }
-            ctx.unsetDeferredSampling();
+    handleUpsampling(ctx: SpanContext, operationName: string, tags: any) {
```

Comment:
Yes, I didn't want to add complexity when we don't have a need for it yet. Do you think it is important to support `allowOnInnerSpans` right now?